src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import exception.ArrayListIndexOutOfBoundsException;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] elements;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= elements.length) {
+            growthElementsArray();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndexExist(index);
+        if (size == elements.length) {
+            growthElementsArray();
+        }
+        System.arraycopy(elements,index,elements,index + 1,
+                elements.length - index - 1);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexExist(index);
+        return elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndexExist(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexExist(index);
+        final T value = elements[index];
+        System.arraycopy(elements,index + 1,elements,index,elements.length - index - 1);
+        size--;
+        return value;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        if (isEmpty()) {
+            throw new ArrayListIndexOutOfBoundsException("Can't remove element from empty list");
+        }
+        int index;
+        for (index = 0; index < elements.length; index++) {
+            if (elements[index] == element
+                    || elements[index] != null && elements[index].equals(element)) {
+                return remove(index);
+            }
+        }
+        throw new NoSuchElementException();
```

Comment:
Exception should be readable
```suggestion
        throw new NoSuchElementException("Element does not exist: " + element);
```